defos/src/defos_html5.cpp

```diff
@@ -0,0 +1,104 @@
+#include <dmsdk/sdk.h>
+#include "defos_private.h"
+
+#if defined(DM_PLATFORM_HTML5)
+
+#include <emscripten.h>
+#include "defos_private.h"
+
+static bool is_maximized = false;
+static WinRect previous_state;
+
+void js_warn(const char* msg) {
+    EM_ASM_({console.warn(UTF8ToString($0))}, msg);
+}
+
+void defos_init() {
+}
+
+void defos_final() {
+}
+
+void defos_event_handler_was_set(DefosEvent event) {
+    js_warn("Callbacks 'on_mouse_leave' and 'on_mouse_enter' don't supported in html5");
+}
+
+void defos_disable_maximize_button() {
+    js_warn("Method 'disable_maximize_button' don't supported in html5");
+}
+
+void defos_disable_minimize_button() {
+    js_warn("Method 'disable_minimize_button' don't supported in html5");
+}
+
+void defos_disable_window_resize() {
+    js_warn("Method 'disable_window_resize' don't supported in html5");
+}
+
+void defos_disable_mouse_cursor() {
+     EM_ASM(Module.canvas.style.cursor = 'none';);
+}
+
+void defos_enable_mouse_cursor() {
+    EM_ASM(Module.canvas.style.cursor = 'default';);
+}
+
+void defos_toggle_fullscreen() {
+    js_warn("Method 'toggle_fullscreen' don't supported in html5, you can use Module.toggleFullscreen() method in JS");
```

Comment:
Why not call `Module.toggleFullscreen()` here?

Reply:
yep, but :
Note: Fullscreen requests need to be called from within an event handler or otherwise they will be denied. 
https://developer.mozilla.org/en-US/docs/Web/API/Fullscreen_API

Reply:
In theory we can use some hack like
Module.canvas.onclick = function()
{
Module.toggleFullscreen();
Module.canvas.onclick = "";
}
But this method would be work only in action.pressed and won't in action.released

Reply:
We could technically queue it to some other event, like mouse moved, or to multiple events, but yeah, it's finicky. Maybe put it on a hidden button and `.click()` it? I'm not sure that would work. I'll try out some things when I come back from work.

Reply:
I checked man variants (even emscripten set*callback and so on) 
I found only one way: 
a hack with on_mouse_up (released) callback, but button should use action.pressed event then.

Reply:
I managed to sort of work around the issue after all. I added an `defos.on_click()` event where the user can manually check if the mouse is clicking on his full screen button and then call `defos.toggle_fullscreen()` from there. This will work because `defos.on_click()` calls its callback inside the native HTML event handler.